module/VuFind/src/VuFind/Session/Redis.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Redis session handler
+ *
+ * Note: Using phpredis extension (see https://github.com/phpredis/phpredis) is
+ * optional, this class use Credis in standalone mode by default
+ *
+ * PHP version 7
+ *
+ * Coypright (C) Moravian Library 2019.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * @category VuFind
+ * @package  Session_Handlers
+ * @author   Veros Kaplan <cpk-dev@mzk.cz>
+ * @author   Josef Moravec <moravec@mzk.cz>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org/wiki/vufind2:session_handlers Wiki
+ */
+namespace VuFind\Session;
+
+/**
+ * Redis session handler
+ *
+ * @category VuFind
+ * @package  Session_Handlers
+ * @author   Veros Kaplan <cpk-dev@mzk.cz>
+ * @author   Josef Moravec <moravec@mzk.cz>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org/wiki/vufind2:session_handlers Wiki
+ */
+class Redis extends AbstractBase
+{
+    /**
+     * Redis connection
+     *
+     * @var \Credis_Client
+     */
+    protected $connection = false;
+
+    /**
+     * Redis version
+     *
+     * @var int
+     */
+    protected $redisVersion = 3;
+
+    /**
+     * Get connection to Redis
+     *
+     * @throws \Exception
+     * @return \Credis_Client
+     */
+    public function getConnection()
```

Comment:
Is there a reason for this to be public rather than protected?

Reply:
The Redis driver is created after `Memcache` session driver, where the `getConnection()` was public.

Reply:
Thanks, @VerosK, that makes sense -- but I have a strong suspicion that the public getConnection in the Memcache driver is an oversight rather than an intentional design choice -- I'll look into that to be sure. In the meantime, it might be worth changing this one if there is no other compelling reason to have it public.

Reply:
I don't know about any other reason for being public. 
So the method `getConnection()` should be probably protected.

Reply:
Agreed.